daemon/indexing/lib/search.js

```diff
@@ -12,11 +12,13 @@ var client = new elasticsearch.Client({
   ]
 })
 
-// Name of the index used for storing Origin data.
-const indexName = 'origin'
-
-// Name of the index type used for storing listings data.
-const listingsType = 'listing'
+// Elasticsearch index and type names for our data
+// Elasticsearch is depreciating storing different types in the same index.
+// (and forbids it unless you enable a special flag)
+const LISTINGS_INDEX = 'listings'
+const LISTINGS_TYPE = 'listing'
+const OFFER_INDEX = 'offers'
```

Comment:
Note that when making the image for the origin-box elasticsearch container, we currently create the "origin" index: https://github.com/OriginProtocol/origin-box/blob/master/dockerfiles/elasticsearch#L11

So we'll need to update that to create "listings" and "offers".

Reply:
Elasticsearch creates the indexes automatically when you first put in a new document into it - it  doesn't seem to require them created ahead of time.

Reply:
To be more specific, what I'm a bit worried about is using dynamic mapping (which we would if we don't create the index) vs explicit mapping. See some [doc](https://www.elastic.co/guide/en/elasticsearch/reference/current/mapping.html) explaining the difference.

I think in the long-term I'd be more comfortable if we would specify an explicit mapping so that we don't run into issues when partners send us garbage docs. Let's discuss...